### PR TITLE
Document the multiple Terraform provider setups

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR adds documentation for new Terraform provider connection methods added as part of RFD 153:

- local temporary bot (`tctl terraform env`)
- native machineID joining (`join_token` + `join_method`)

This also describes the two existing setups:
- with tbot (current docs)
- with static creds (from v14 docs)

### CODE REVIEW

File: docs/pages/management/dynamic-resources/terraform-import-existing-resources.mdx
Comment (hugoShaka) [inline docs/pages/management/dynamic-resources/terraform-import-existing-resources.mdx:1]: This page is missing an entry in the config.json

File: docs/pages/management/dynamic-resources/terraform-provider.mdx [hunk 1]
```diff
@@ -1,291 +1,33 @@
...
-GitOps, read the guide to using the Teleport Terraform provider with [Spacelift
-and Machine ID](../../enroll-resources/machine-id/deployment/spacelift.mdx).
+| Guide                                                                                                               | Use-case                                                                                                                                                                                                         |
+|---------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [Run the Teleport Terraform provider locally](./terraform-provider/local.mdx)                                       | This is the best way to get started with the Teleport terraform provider, write some initial Terraform code and achieve IaC.                                                                                     |
```
[the remark this answers is not in the document]
Comment (hugoShaka) [reply docs/pages/management/dynamic-resources/terraform-provider.mdx:16]: I think the other guides are not intended for initial terraform development. The feedback loop is too slow and painful. I will rephrase.

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx [hunk 2]
```diff
@@ -0,0 +1,394 @@
+---
+title: Run the Teleport Terraform Provider in CI or Cloud
+description: How to manage dynamic resources using the Teleport Terraform provider from your CI pipelines or Cloud provider.
+---
+
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx:5]: I would add a brief intro paragraph here that explains the scope of this guide so a user knows whether to keep reading.

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+join method and token to the provider.
+
+- [Azure MachineID guide](../../../enroll-resources/machine-id/deployment/azure.mdx)
+- [Kubernetes MachineID guide](../../../enroll-resources/machine-id/deployment/kubernetes.mdx)
+- [TPM MachineID guide](../../../enroll-resources/machine-id/deployment/linux-tpm.mdx)
+
+</Admonition>
+
+This guide does not cover running Teleport locally or on a dedicated server.
+If you are in one of those cases, please follow the dedicated guides:
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx:42]:
```suggestion
If you are in one of those cases, follow the dedicated guides:
```

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+    allow:
+    - project_ids:
+        - "<Var name="my-project-123456" />"
+      service_accounts:
+        - "<Var name="my-service-account" />@<Var name="my-project-123456" />.iam.gserviceaccount.com"
+```
+</TabItem>
+</Tabs>
+
+### Step 3/4. Configure your terraform provider
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/ci-or-cloud.mdx:199]:
```suggestion
## Step 3/4. Configure your Terraform provider
```

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+Validate that you meet the version requirements (16.2 or greater) by running:
+
+```code
+$ tsh status
+Teleport v(=teleport.version=) go(=teleport.golang=)
+Proxy version: (=teleport.version=)
+Proxy: <Var name="teleport.example.com:443" />
+```
+
+### Step 1/2. Generate temporary bot credentials
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx:51]:
```suggestion
## Step 1/2. Generate temporary bot credentials
```

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+This is an admin-level action and requires MFA to complete
+Tap any security key
+Detected security key tap
+⚙️ Creating temporary bot "tctl-terraform-env-82ab1a2e" and its token
+🤖 Using the temporary bot to obtain certificates
+🚀 Certificates obtained, you can now use Terraform in this terminal for 1h0m0s
+```
+
+
+### Step 2/2. Run the terraform provider
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx:75]:
```suggestion
## Step 2/2. Run the Terraform provider
```

File: docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+   
+   Plan: 1 to add, 0 to change, 0 to destroy.
+   ```
+
+If the plan succeeds, the Terraform provider successfully connected to Teleport.
+You can now start developing locally with the Teleport Terraform provider.
+
+Do not forget to obtain new temporary credentials every hour by re-running `eval $(tctl terraform env)`.
+
+### Next steps
```
Comment (ptgott) [inline docs/pages/admin-guides/infrastructure-as-code/terraform-provider/local.mdx:157]:
```suggestion
## Next steps
```

**Review by ptgott — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**